capsules/src/usb/usb_user.rs

```diff
@@ -135,25 +139,24 @@ where
                     .enter(appid, |app, _| {
                         if app.awaiting.is_some() {
                             // Each app may make only one request at a time
-                            ReturnCode::EBUSY
+                            Err(ErrorCode::BUSY)
                         } else {
-                            if app.callback.is_some() {
-                                app.awaiting = Some(Request::EnableAndAttach);
-                                ReturnCode::SUCCESS
-                            } else {
-                                ReturnCode::EINVAL
-                            }
+                            app.awaiting = Some(Request::EnableAndAttach);
+                            Ok(())
```

Comment:
Flagging this as it's a change in behavior (maybe okay): This used to reject attachment if a callback hadn't been set yet; now it will allow it. I think the original behavior could be retained by checking what the callback is set to.

Reply:
I think there was an explicit decision not to allow capsules to make decisions on the basis of whether a callback has been set. See the discussion here https://github.com/tock/tock/pull/2258#pullrequestreview-551879183